src/Mvc/Mvc.Core/test/FileContentResultTest.cs

```diff
@@ -71,8 +72,11 @@ public void Constructor_SetsLastModifiedAndEtag()
             MediaTypeAssert.Equal(expectedMediaType, result.ContentType);
         }
 
-        [Fact]
-        public async Task WriteFileAsync_CopiesBuffer_ToOutputStream()
+        [Theory]
+        [MemberData(nameof(GetActions))]
```

Comment:
I know this sounds onerous, but could we split these apart as separate tests? We don't want TheoryData as a way to drive different test code paths. It makes it harder to diagnose, skip individual scenarios that have bugs etc.

Reply:
Sure, that make sense, I will add a new TestFile and duplicate de tests calling IResult.ExecuteAsync instead, thanks!

Reply:
Can we share tests with a base class or is that not possible?

Reply:
Sure, I was planning to create base clases but for share the infrastructure for the test than to share the test itself, I am going to explore that idea, I can possibly have the methods extracted in a base clase and call those in the implementations.

Reply:
@pranavkm @davidfowl I have used some static methods to share the implementation of the test but the tests itself are separated and in different files, let me know if that is acceptable or if I should simple rollback to separate files and duplicate test. thanks!